src/sentry/integrations/github/webhook.py

```diff
@@ -92,7 +93,28 @@ def __call__(self, event: Mapping[str, Any], host: str | None = None) -> None:
                 organization_id__in=orgs.keys(),
                 provider=f"integrations:{self.provider}",
                 external_id=str(event["repository"]["id"]),
-            )
+            ).exclude(status=ObjectStatus.HIDDEN)
+
+            if not repos.exists():
+                binding_key = "integration-repository.provider"
+                provider_key = "integrations:" + integration.provider
+                provider_cls = bindings.get(binding_key).get(provider_key)
+                provider = provider_cls(id=provider_key)
+
+                config = {
+                    "integration_id": integration.id,
+                    "external_id": str(event["repository"]["id"]),
```

Comment:
`event["repository"]["id"]`

is there guaranteed that `event` will always have a 'repository'?
maybe do 
`event.get("repository", {}).get("id")` instead?

Reply:
https://github.com/getsentry/sentry/pull/52756/files#diff-b742e5d44db73d9f58ad922d7347c4982d624e58e04a3dd73916cc558c4be952R83 ensures that repository will always be in event